arangod/Aql/BindParameters.cpp

```diff
@@ -73,7 +73,7 @@ void BindParameters::process() {
       THROW_ARANGO_EXCEPTION_PARAMS(TRI_ERROR_QUERY_BIND_PARAMETER_TYPE, key.c_str());
     }
 
-    _parameters.emplace(std::move(key), std::make_pair(value, false));
+    _parameters.try_emplace(key, value, false);
```

Comment:
we are not moving `key` anymore. why?

Reply:
because the first parameter is a `const&`

Reply:
According to the C++ standard there should also be an overload that takes a universal reference (https://en.cppreference.com/w/cpp/container/map/try_emplace).

Reply:
I was on that page an overlooked it:( And indeed the promise for the value seems to apply to the key as well! http://eel.is/c++draft/unord.map#modifiers-10

Reply:
Therefore even moving the key in a loop is ok - awesome!